packages/controllers/src/snaps/SnapController.test.ts

```diff
@@ -1148,6 +1175,121 @@ describe('SnapController', () => {
     await service.terminateAllSnaps();
   });
 
+  it('does not timeout while waiting for response from MetaMask', async () => {
+    const [snapController, service] = getSnapControllerWithEES(
+      getSnapControllerWithEESOptions({
+        idleTimeCheckInterval: 30000,
+        maxIdleTime: 160000,
+      }),
+    );
+    const sourceCode = `
+    module.exports.onRpcRequest = () => wallet.request({ method: 'eth_blockNumber', params: [] });
+    `;
+
+    const snap = await snapController.add({
+      origin: MOCK_ORIGIN,
+      id: MOCK_SNAP_ID,
+      sourceCode,
+      manifest: getSnapManifest({ shasum: getSnapSourceShasum(sourceCode) }),
+    });
+
+    jest
+      // Cast because we are mocking a private property
+      .spyOn(service, 'setupSnapProvider' as any)
+      .mockImplementation((_snapId, rpcStream) => {
+        const mux = setupMultiplex(rpcStream as Duplex, 'foo');
+        const stream = mux.createStream('metamask-provider');
+        const engine = new JsonRpcEngine();
+        const middleware = createAsyncMiddleware(async (req, res, _next) => {
+          if (req.method === 'metamask_getProviderState') {
+            res.result = { isUnlocked: false, accounts: [] };
+          } else if (req.method === 'eth_blockNumber') {
+            await new Promise((resolve) => setTimeout(resolve, 400));
```

Comment:
We can now (after removing Electron) use fake timers in all `SnapController` tests but it may be better to create a task to use fake timers in all tests? Since otherwise this PR will scope-creep.

Reply:
I'd say better to add to one/two tests now and add more bit by bit than doing a big task that might never be completed

Reply:
You can create an issue an assign it to me then, I'll stick my neck out on this one 😅